### DISCUSSION TITLE
Libvirt connection from a controller in a container does not work

### DISCUSSION THREAD

**Discussion by ezr-ondrej** [foresterorg/forester#4] [General]
I don't seem to understand, how the connection to libvirt socket should work from the container? :thinking:

Do I need to share the socket?

Also, the socket is hardcoded to `/var/run/libvirt/libvirt-sock`, but libvirt have split the socket into multiple and this is what I've got on Fedora 38:
```
> ls /var/run/libvirt/ | grep sock
libvirt-admin-sock
libvirt-sock-ro
virtinterfaced-sock
virtlockd-sock
virtlogd-sock
virtnetworkd-admin-sock
virtnetworkd-sock
virtnetworkd-sock-ro
virtnodedevd-admin-sock
virtnodedevd-sock
virtnodedevd-sock-ro
virtnwfilterd-sock
virtqemud-admin-sock
virtqemud-sock
virtqemud-sock-ro
virtsecretd-sock
virtstoraged-admin-sock
virtstoraged-sock
virtstoraged-sock-ro
```

I'd document how would one pass the `-u uri` param, tho I guess it would still not work from the container.

**Comment by lzap** [comment]
Update to a0f21507 and use `tcp://hostname:16509` as a URI.

https://lukas.zapletalovi.com/posts/2023/enable-libvirt-over-tcp/
